doc/api_reference.md

```diff
@@ -92,13 +92,21 @@ Returns a sender that transforms the value of the `predecessor` by calling
 Returns a sender that will first launch `source` and upon completion of
 `source` will launch the `completion` sender.
 
-If `completion` completes with `set_value()` (which must ) then the
-composed operation completes with the result of `source`.
-Otherwise, if `completion` sender completes with 'done' or 'error'
+If `completion` completes with `set_value()` (which must complete with an
+empty value pack) then the composed operation completes with the result of
+`source`.
+Otherwise, if `completion` sender completes with `set_done` or `set_error`
 then the composed operation completes with the result of `completion`.
 
-Note that `completion` sender must complete with an empty value pack.
-ie. be a `void`-value sender.
+The composed finally-operation will complete inline on the execution context
+that the `completion` sender completes on, except in the case that the call
+to `connect()` on the completion-sender exits with an exception, in which case
+the operation will complete with `set_error()` inline on whatever execution
+context the `source` sender completed on.
+
```

Comment:
> on whatever execution
context the `source` sender completed on

I wonder how finally and on should compose with a scheduler that guarantees completion on a defined context. It would be surprising if they were able to thwart the scheduler contract, but perhaps that is impossible to maintain..